### PR TITLE
Vbt footer

### PR DESCRIPTION
**Pull request by deadlight**
## Done

Converted the footer to Vanilla brochure theme.

## QA

Test regressions:
 - ./run
 - Make sure that the site looks no different to live

Test new features:
 - Change a template to use the v1 base template (e.g. `{% extends "about/base_about-v1.html" %}`)
 - Navigate to the changed page and test the footer in all viewports - it should be as identical to live as possible.

Related #1670

### CODE REVIEW

File: static/sass/_patterns_footer.scss [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+      margin-top: 10px;
+    }
+  }
+
+  .p-inline-list,
+  .p-inline-list--small,
+  .p-inline-list--middot-small {
+    margin-left: 0;
+    margin-top: 0;
+    padding-left: 0;
```
Comment (anthonydillon) [inline static/sass/_patterns_footer.scss]: These patterns already have padding and margin left set to 0 in the pattern themselfs.

File: static/sass/styles-v1.scss [hunk 2]
```diff
@@ -12,6 +12,13 @@
 // vendor
 @import 'vendors/global-nav/build';
 
+// patterns
```
Comment (anthonydillon) [inline static/sass/styles-v1.scss]: What's the difference between this file and the one if the CSS folder?

File: templates/templates/footer-v1.html [hunk 3]
```diff
@@ -0,0 +1,124 @@
+{% block footer %}
+<footer class="p-footer p-strip u-clearfix">
+  <div class="row p-footer__container">
+    <p class="u-hidden--medium u-hidden--large link-to-top"><a href="#"><small>Back to top</small></a></p>
+    <nav id="main-navigation" class="p-footer__nav u-clearfix">
+      <div class="strip-inner-wrapper strip-inner-wrapper--nav">
```
Comment (anthonydillon) [inline templates/templates/footer-v1.html]: Does strip-inner-wrapper do anything here?

File: templates/templates/footer-v1.html [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        <div class="p-footer__nav-col col-2 u-no-margin--bottom">
+          <ul class="p-footer__links">
+            <li class="p-footer__item"><h2 class="p-footer__title"><a class="p-link--soft" href="/cloud">Cloud</a></h2>
```
Comment (anthonydillon) [inline templates/templates/footer-v1.html]: Please bring down the h2 and children on to the own line.

File: static/sass/_patterns_footer.scss [hunk 5]
```diff
@@ -0,0 +1,234 @@
+// Footer pattern styling
+
+$color-shadow: rgba(0, 0, 0, .3);
+
+@mixin footer-title {
+  color: $color-mid-dark;
+  cursor: pointer;
+  line-height: 1.5em;
```
Comment (barrymcgee) [inline static/sass/_patterns_footer.scss]: Line height should not need a unit.

File: static/sass/_patterns_footer.scss [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+  margin-bottom: 0;
+}
+
+@mixin ubuntu-p-footer {
+  .p-footer {
+    border: 0;
+    box-shadow: 0 4px 4px -4px $color-shadow inset;
+    font-size: 1rem;
```
Comment (barrymcgee) [inline static/sass/_patterns_footer.scss]: Can you use Vanilla spacing units please?

File: static/sass/_patterns_footer.scss [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+      @include footer-title;
+      background-image: url("#{$assets-path}e4a5a8c0-picto-upload-warmgrey.svg");
+      background-position: 10px center;
+      background-repeat: no-repeat;
+      background-size: 14px 14px;
+      border-bottom: 1px solid $color-mid-dark;
+      display: block;
+      left: auto;
+      margin-bottom: -1px;
+      padding: 10px 0 10px 30px;
```
Comment (barrymcgee) [inline static/sass/_patterns_footer.scss]: I would recommend not rems but Vanilla spacing variable - in cases where there is not 1:1 match but they are only a couple of pixels out and doesn't break anything, use the spacing variable. Remember we have agreed with design that this migration will not be a pixel perfect representation, the whole point of using a framework is to let the framework do the heavy lifting.

File: static/sass/styles-v1.scss (hunk 2 above)
Comment (barrymcgee) [inline static/sass/styles-v1.scss]: We don't want to couple old CSS and v1 CSS - the last step in this migration will be to nuke all old styles without having to worry about dependencies.

Comment (anthonydillon) [inline static/sass/styles-v1.scss]: What confuses me is that both files are called `styles-v1.scss` just ones in a folder called css and the other in a sass folder.

**Comment by anthonydillon** [comment]
QA issues:
- The contact us button has an underline when hovered. This could be fixed by using a vanilla button pattern.
- The switch to mobile styling is at a different breakpoint but seems ok apart from at 780px which has the global nav:
about the ubuntu project - ubuntu 4

File: static/sass/styles-v1.scss (hunk 2 above)
Comment (deadlight) [inline static/sass/styles-v1.scss]: It's just an artefact of the code restructure that happened after this branch was made, I'll remove the old one.

Comment (barrymcgee) [inline static/sass/styles-v1.scss]: @deadlight Careful - non-v1 pages still rely on the old one..

File: static/sass/_patterns_footer.scss [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+  $color-shadow: rgba(0, 0, 0, .3);
+
+  .p-footer {
+    border: 0;
+    box-shadow: 0 4px 4px -4px $color-shadow inset;
+    margin-bottom: $sp-medium;
+    padding-top: 0;
+
+    @media only screen and (min-width: $breakpoint-medium) {
+      padding-top: .75rem;
```
Comment (barrymcgee) [inline static/sass/_patterns_footer.scss]: `$sp-small`

File: static/sass/_patterns_footer.scss [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+      margin: 0;
+      padding: 0;
+    }
+
+    &__item {
+      display: block;
+      margin-bottom: 0;
+
+      @media only screen and (min-width: $breakpoint-medium + 1) {
+        margin-bottom: 1.5rem;
```
Comment (barrymcgee) [inline static/sass/_patterns_footer.scss]: `$sp-large`

File: static/sass/_patterns_footer.scss [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+            font-size: .8125rem;
+            padding: 0 0 $sp-x-small;;
+          }
+        }
+      }
+    }
+  }
+
+  .p-footer--secondary {
+    padding: 20px 10px 0;
```
Comment (barrymcgee) [inline static/sass/_patterns_footer.scss]: Spacing variables?

File: static/sass/_patterns_lists.scss [hunk 11]
```diff
@@ -0,0 +1,114 @@
+// Site specific list styling
+
+@mixin list-link {
+  display: block;
+  height: 32px;
+  fill: $color-mid-dark;
+  width: 32px;
```
Comment (barrymcgee) [inline static/sass/_patterns_lists.scss]: Spacing variables?

File: static/sass/_patterns_footer.scss [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+      }
+
+      @media only screen and (min-width: $breakpoint-large) {
+        font-size: .8125rem;
+      }
+
+      a:link,
+      a:visited {
+        color: $color-mid-dark;
+        padding-left: 10px;
```
Comment (barrymcgee) [inline static/sass/_patterns_footer.scss]: `$sp-x-small` ?

Comment (deadlight) [inline static/sass/_patterns_footer.scss]: damn it!

**Comment by barrymcgee** [comment]
@deadlight Looks good to me, resolve conflicts and merge if @anthonydillon's issues have also been addressed

**Pull request merged**